java/arcs/core/storage/CapabilitiesResolver.kt

```diff
@@ -70,63 +76,59 @@ class CapabilitiesResolver(
         // TODO: This is a naive and basic solution for picking the appropriate
         // storage key creator for the given capabilities. As more capabilities are
         // added the heuristics will become more robust.
-        val protocols = findStorageKeyProtocols(capabilities)
-        require(protocols.isNotEmpty()) {
-            "Cannot create. a suitable storage key for $capabilities"
+        val creators = findCreators(capabilities)
+        require(creators.isNotEmpty()) {
+            "Cannot create a suitable storage key for $capabilities"
         }
-        if (protocols.size > 1) {
+        if (creators.size > 1) {
             log.warning { "Multiple storage key creators for $capabilities" }
         }
-        val creator = requireNotNull(creators[protocols.first()]?.second)
-        val backingKey = creator.invoke(BackingStorageKeyOptions(options.arcId, entitySchema))
-        val storageKey = creator.invoke(ContainerStorageKeyOptions(options.arcId, entitySchema))
+        val create = creators.first().create
+        val backingKey = create.invoke(BackingStorageKeyOptions(options.arcId, entitySchema))
```

Comment:
I *think* you can just use `create(...)` instead of using `invoke`

Reply:
you're right!!! (at some point something didn't work and i ended up with `invoke`, but now it works, so .. gone)